packages/next/server/app-render.tsx

```diff
@@ -138,11 +139,41 @@ function useFlightResponse(
           rscCache.delete(id)
           writer.close()
         } else {
+          const responsePartial = decodeText(value)
+          const css = responsePartial
+            .split('\n')
+            .map((partialLine) => {
+              const line = remainingFlightResponse + partialLine
+              remainingFlightResponse = ''
+
+              try {
+                const match = line.match(/^M\d+:(.+)/)
+                if (match) {
+                  return JSON.parse(match[1])
+                    .chunks.filter((chunkId: string) =>
+                      chunkId.endsWith('.css')
+                    )
+                    .map(
+                      (file: string) =>
+                        `<link rel="stylesheet" href="/_next/${file}">`
+                    )
+                    .join('')
+                }
+                return ''
+              } catch (err) {
+                // The JSON is partial
+                remainingFlightResponse = line
+                return ''
+              }
```

Comment:
rather than overparsing why not rewrite this a while loop that looks for the next line break. only process if there is a next line break.

Here is basically the same algo in React: https://github.com/facebook/react/blob/main/packages/react-client/src/ReactFlightClientStream.js#L100-L109